components/abstractions/src/main/java/com/microsoft/kiota/RequestInformation.java

```diff
@@ -268,26 +271,31 @@ public void setResponseHandler(@Nonnull ResponseHandler responseHandler) {
     /**
      * Sets the request body to be a binary stream.
      * @param value the binary stream
+     * @throws IOException when the stream cannot be read.
      * @deprecated use {@link #setStreamContent(InputStream, String)} instead.
      */
     @Deprecated
-    public void setStreamContent(@Nonnull final InputStream value) {
+    public void setStreamContent(@Nonnull final InputStream value) throws IOException {
```

Comment:
Adding new checked exceptions will be source breaking for the generated code.
(Requiring the exception to be handled, or be declared at a higher level)
We should probably wrap this to maintain compatibility.
@andreaTP to provide additional input if he can

Reply:
I read in _Effective Java_ that checked exceptions should be used for recoverable errors and runtime exceptions for programming errors. I think failing to set the stream content may not be recoverable for that specific request.

Reply:
yes, ideally we would have set this checked exception in the method prototype before the GA.
but since we GAed the language, adding this now would be a source breaking change.
i.e.: any client that's been generated to date will break if they pull the latest kiota dependencies without re-generating the client.